src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateDtoValidator.java

```diff
@@ -93,25 +101,53 @@ protected void validateAttributeAndNameCodes(final BindingResult errors, final L
 		germplasmUpdateDTOList.stream().filter(germ -> germ.getAttributes() != null).collect(Collectors.toList())
 			.forEach(
 				g -> attributesCodes.addAll(g.getAttributes().keySet().stream().map(n -> n.toUpperCase()).collect(Collectors.toList())));
-		final Set<String> existingAttributesCodes =
-			this.germplasmAttributeService.filterGermplasmAttributes(attributesCodes, null).stream().map(AttributeDTO::getCode).collect(
-				Collectors.toSet());
 
-		if (!nameCodes.equals(existingNamesCodes)) {
+		final VariableFilter variableFilter = new VariableFilter();
```

Comment:
Same comment as in Commons (applicable to Middleware, Workbench too), consider a Middleware call to expose a service to get all attribute and passport variables

Reply:
I think we should have VariableService in middleware to contains everything related to variable. I would not create a separated one for attributes and passport exclusively
I think we can create a ticket for that because we already did lot of changes as part of this ticket.
@darla-leafnode